deployments/clowder.yaml

```diff
@@ -188,3 +188,8 @@ parameters:
     description: |
       It allows to validate API requests by using the
       service OpenAPI specification.
+  # TODO HMS-3528 Set "true" or remove the feature flag
+  - name: APP_ENABLE_RBAC
+    value: "false"
+    description: |
+      It allows to enable / disable RBAC middleware.
```

Comment:
I think this should be set to `true` by default. The app-interface change will set it to `false` for stage and ephemeral. You also have to pass the parameter as env var to the container.

Reply:
or we just remove the definition (now or later) - absence shall imply "enabled".

Reply:
Indeed, I recognize it is confusing; my idea was "enabled by default" when everything is finished, but at the same time, don't brake anything until it is ready to work, so:

- When nothing is specified, the default value in the code (configuration) is enabled.
- In the config.yaml it is set the value as `false`, so it does not impact to the developer in other areas.
- In the makefile it is defined optional to `false`, to make possible to override the value from the CLI `APP_ENABLE_RBAC=true make run`, or the opposite.
- In the `clowder.yaml`, I had to define the parameter to be used from app-interface; it is useful when deploying in ephemeral so nothing is broken in IQE.
- In the app-interface, I force to false (until HMS-3528 is merged that enable everything); probably this is where no change is necessary at all; but when I thought about it at the beginning it made sense.

Reply:
my conclusion; if we set this `true`, we will brake the IQE tests until the middleware is implemented (bear in mind they deploy by `bonfire deploy [--frontends true] idmsvc` with no additional parameters (which is right; what is in the makefiles is to make the developers life easier so the bonfire options are automatically filled to get the current commit state deployed); IMHO I would leave this parameter as it is.

Another question is if the app-interface MR is necessary; because we are managing the dafault value in the openshift template until we get the middleware ready. I think it could be cancelled.

What do you think?

Reply:
Another thought in relation with the app-interface: from a perspective of a gradual deployment (1) change the deployment parameter in ephemeral (or remove), and when the feature looks working correctly, (2) change the deployment parameter (or remove) in stage, and get everything working.

So `app-interface` MR is necessary too.

Reply:
The app-interface MR has landed